test/util/util.go

```diff
@@ -0,0 +1,346 @@
+/*
+Copyright 2023 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	ocmWorkClient "open-cluster-management.io/api/client/work/clientset/versioned"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	ksapi "github.com/kubestellar/kubestellar/api/control/v1alpha1"
+	ksClient "github.com/kubestellar/kubestellar/pkg/generated/clientset/versioned"
+)
+
+const (
+	timeout = 500 * time.Second
+)
+
+func GetConfig(context string) *rest.Config {
+	config, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		&clientcmd.ClientConfigLoadingRules{ExplicitPath: os.Getenv("KUBECONFIG")},
+		&clientcmd.ConfigOverrides{CurrentContext: context}).ClientConfig()
+	gomega.ExpectWithOffset(1, err).NotTo(gomega.HaveOccurred())
+	return config
+}
+
+func CreateKubeClient(config *rest.Config) *kubernetes.Clientset {
+	clientset, err := kubernetes.NewForConfig(config)
+	gomega.ExpectWithOffset(1, err).NotTo(gomega.HaveOccurred())
+	return clientset
+}
+
+func CreateKSClient(config *rest.Config) *ksClient.Clientset {
+	clientset, err := ksClient.NewForConfig(config)
+	gomega.ExpectWithOffset(1, err).NotTo(gomega.HaveOccurred())
+	return clientset
+}
+
+func CreateOcmWorkClient(config *rest.Config) *ocmWorkClient.Clientset {
+	ocmWorkClient, err := ocmWorkClient.NewForConfig(config)
+	gomega.ExpectWithOffset(1, err).NotTo(gomega.HaveOccurred())
+	return ocmWorkClient
+}
+
+func CreateDynamicClient(config *rest.Config) *dynamic.DynamicClient {
+	dynamicClient, err := dynamic.NewForConfig(config)
+	gomega.ExpectWithOffset(1, err).NotTo(gomega.HaveOccurred())
+	return dynamicClient
+}
+
+func CreateNS(ctx context.Context, client *kubernetes.Clientset, name string) {
+	labels := make(map[string]string)
+	labels["app.kubernetes.io/name"] = "nginx"
+	nsObj := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: "",
+			Labels:    labels,
+		},
+		Status: corev1.NamespaceStatus{},
```

Comment:
FYI: I do not think you need this line.
See https://github.com/kubernetes/api/blob/v0.28.2/core/v1/types.go#L5831

Reply:
removed.